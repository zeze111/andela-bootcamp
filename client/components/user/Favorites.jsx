import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 *
 *
 * @class Favorites
 * @extends {React.Component}
 */
class Favorites extends Component {
  /**
   * @param {any} event
   * @memberof Home
   * @return {void}
   */
  clickEvent = () => {
    this.props.deleteFavorite(this.props.favorite.id);
  }

  /**
   * @memberof Home
   * @return {void}
   */
  render() {
    const { favorite } = this.props;

    return (
      <div>
        <li className="collection-item">
          <Link
            to={`/userRecipe/${favorite.id}`}
            href={`/userRecipe/${favorite.id}`}
            className="caps text-color"
          >
            {favorite.name}
            <span className="black-text text-space">
              posted by {favorite.User.firstName} {favorite.User.surname}
            </span>
          </Link>
          <div className="secondary-content div-pointer2">
            <i
              role="button"
              tabIndex="0"
              onKeyDown={this.handleKeyDown}
              onClick={this.clickEvent}
              className="material-icons icon-color"
            > delete
            </i>
          </div>
        </li>
      </div>
    );
  }
}

Favorites.propTypes = {
  deleteFavorite: PropTypes.func.isRequired,
  favorite: PropTypes.objectOf(PropTypes.any)
};

Favorites.defaultProps = {
  favorite: {}
};

export default Favorites;
