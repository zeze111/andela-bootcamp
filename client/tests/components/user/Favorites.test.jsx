import React from 'react';

import Favorites from '../../../components/user/Favorites';
import store from '../../../store';

const props = {
  deleteFavorite: jest.fn(() => Promise.resolve()),
  favorite: {
    id: 1,
    name: 'blah',
    User: {
      id: 20,
      firstName: 'Jane'
    }
  },
  isLoading: false
};

const setup = () => {
  const wrapper = shallow(<Favorites {...props} store={store} />);
  return { wrapper }
}

describe('clickEvent', () => {
  it('should be called on delete a review', () => {
    const { wrapper } = setup();

    wrapper.instance().clickEvent();
  });
});

describe('Test for Favorites', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
