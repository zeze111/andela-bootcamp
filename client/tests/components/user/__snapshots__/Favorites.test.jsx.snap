// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Favorites should render component 1`] = `
Object {
  "wrapper": <div>
    <li
      className="collection-item list"
    >
      <Link
        className="caps text-color break"
        href="/recipe/1"
        replace={false}
        to="/recipe/1"
      >
        blah
        <span
          className="black-text text-space"
        >
          posted by 
          Jane
           
        </span>
      </Link>
      <div
        className="secondary-content div-pointer2 "
      >
        <i
          className="material-icons icon-color"
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
           delete
        </i>
      </div>
    </li>
  </div>,
}
`;
